python/hail/methods/statgen.py

```diff
@@ -1100,3 +1100,192 @@ def rrm(call_expr):
                                             rrm,
                                             [row.s for row in dataset.cols_table().select('s').collect()],
                                             n_variants)
+
+class FilterAlleles(object):
+    """Filter out a set of alternate alleles.  If all alternate alleles of
+    a variant are filtered out, the variant itself is filtered out.
+    `filter_expr` is an alternate allele indexed `Array[Boolean]`
+    where the booleans, combined with `keep`, determine which
+    alternate alleles to filter out.
+
+    This object has bindings for values used in the filter alleles
+    process: `old_to_new`, `new_to_old` and `new_v`.  Call
+    :meth:`.FilterAlleles.annotate_rows` and/or
+    :meth:`.FilterAlleles.annotate_entries` to update row-indexed and
+    row- and column-indexed fields for the new, allele filtered
+    variant.  Finally, call :meth:`.FilterAlleles.filter` to perform
+    filter alleles.
+
+    Examples
+    --------
+    
+    Filter alleles with zero AC count on a dataset with the HTS entry
+    schema and update the ``info.AC`` and entry fields.
+
+    >>> fa = methods.FilterAlleles(dataset.info.AC.map(lambda AC: AC == 0), keep=False)
```

Comment:
How do you feel about providing `methods.filter_alleles` as a function which constructs a FilterAlleles object?

Reply:
So I wrote that first.  I admit, it looks a bit better, but I tried this because (1) the difference in syntax signals it isn't a normal method call (but a multi-part call), and (2) breaking up the docs between the method call and the returned object was a bit annoying.

I can change it back if you prefer.

Reply:
This is fine. Let's get some feedback from Konrad/others and go from there.